packages/react-query/src/shared/proxy/utilsProxy.ts

```diff
@@ -265,6 +292,18 @@ export function createReactQueryUtilsProxy<
           context.setInfiniteQueryData(queryKey, updater, ...rest),
         getData: () => context.getQueryData(queryKey),
         getInfiniteData: () => context.getInfiniteQueryData(queryKey),
+        getQueryOptions: () => {
```

Comment:
this is nice 🙌 . I'm just wondering if this somehow overlaps with:

https://github.com/trpc/trpc/pull/3302/files#diff-ef4e1d08a351372bae0fff79abe60481bb5e00914f176984386eaa48c1e209bdR288

which exposes a function to get the `queryKey` only.

If we have a function that returns all options, wouldn't the one for the `queryKey` only be a bit unnecessary, as you could just as well do:

```
utils.post.byId.getQueryOptions({ id: '1' }).queryKey
```

or maybe the extra function is fine, but it could be implemented with that internally - or is this different?